src/lib.rs

```diff
@@ -2051,10 +2051,8 @@ where
             Some(len) => len,
             None => return None,
         };
-        if bytes.len() < expected_len {
-            return None;
-        }
-        let (bytes, suffix) = bytes.split_at(expected_len);
+        let position = bytes.len().checked_sub(expected_len)?;
+        let (bytes, suffix) = bytes.split_at(position);
```

Comment:
To be consistent with the implementation of `new_from_suffix`:

```suggestion
        let split_at = bytes.len().checked_sub(expected_len)?;
        let (bytes, suffix) = bytes.split_at(split_at);
```